### DISCUSSION TITLE
Multiple root layouts and root `not-found`?

### DISCUSSION THREAD

**Discussion by victorandree** [vercel/next.js#50034] [App Router]
> [!NOTE]
> As of May 2024, @victorandree (the author) hasn't been testing this issue or listed workarounds for newer versions of Next.js in a while. Please refer to the replies for up-to-date information.

Multiple root layouts are a highlighted feature of the ["Route Groups" page in the Next.js documentation](https://nextjs.org/docs/app/building-your-application/routing/route-groups). On the other hand, [a "root `app/not-found.js`" file is required to handle "global" 404 errors](https://nextjs.org/docs/app/api-reference/file-conventions/not-found). This global `not-found.js` requires a single root layout.

How can we then support a global 404 page while using route groups to create multiple root layouts? I don't think we can.

- **Feature request:** Unlike `app/global-error.js`, which handles "global errors", the `app/not-found.js` requires a root layout. IMO, there should be a `app/global-not-found.js` that works in a similar way. Or, to avoid another file convention, a way to handle `notFound` errors in a normal error handler.
- **Workaround:** There's 2 possible workarounds:

  - Dynamic catch-all route: Create a `(app1)/[...not-found]/page.tsx` page that calls `notFound()`. This will trigger the the closest `not-found.tsx` to render. Thanks @ginifizz!
  - Original: As a (hopefully temporary) workaround, I'm keeping my global not found in `pages/404.js` for now.

## App directory layouts

Basic setup where you have a public website and admin section, where the admin section uses a different root layout (maybe to skip on analytics, use an enterprise CSS framework):

```plain
app
├── (app1)
│   ├── layout.tsx
│   └── page.tsx
└── (admin)
    └── admin
        ├── layout.tsx
        └── page.tsx
```

- Adding `app/not-found.tsx` to the root directory does not work. It'll crash with this message:

  > `app/not-found.tsx`
  > not-found.tsx doesn't have a root layout. To fix this error, make sure every page has a root layout.

- Adding an `app/layout.tsx` will break things in strange ways.

- Adding `not-found.tsx` to the route groups (e.g. `(admin)/not-found.tsx`) will simply not work. The pages aren't found.

## Workaround with dynamic catch-all route

> [!WARNING]
> This workaround is broken in 13.4.10 through 13.4.13. It works in newer versions of Next.js. The `(app1)/not-found.tsx` page will render without any root layout (i.e. with an "empty" root layout).

You can add a dynamic catch-all route to catch all and in this route you can call `notFound`. This triggers the closest `not-found.tsx`.

```plain
app
├── (app1)
│   ├── [...not-found]
│   │   └── page.tsx
│   ├── layout.tsx
│   ├── not-found.tsx
│   └── page.tsx
└── (app2)
    ├── admin
    │   └── page.tsx
    └── layout.tsx
```

In `(app1)/[...not-found]/page.tsx`:

```typescript
import { notFound } from 'next/navigation';

export default function NotFoundDummy() {
  notFound();
}
```

(You want to actually call `notFound()` here and not just render a 404, in order for the 404 status to be set correctly).

**Comment by maxschulmeister** [comment]
lol I am trying to solve this exact problem currently. +1 for this

**Comment by silkquasar** [comment]
In the root of your app with the not-found.tsx  use global-error.tsx instead of layout.tsx.  The global-error.tsx uses html and body tags just as layout.tsx.  This should allow use of non-found.tsx in the root directory without a layout.tsx.

**Reply by silkquasar** [reply]
> In the root of your app with the not-found.tsx use global-error.tsx instead of layout.tsx. The global-error.tsx uses html and body tags just as layout.tsx. This should allow use of non-found.tsx in the root directory without a layout.tsx.

Sorry.  I decided to test.  The error is removed in development.  However, the problem returns when trying to build a app.

**Comment by Daniel4lm** [comment]
This is still an issue in Next.js 13.4.4. I hope they will find some better solution for this.

**Reply by ilyahorski** [reply]
Hi, try adding a layout.tsx file in the directory to your not-found.tsx file. It's work for me.
`layout.tsx`
```
import { ReactNode } from 'react';

interface ErrorProps {
  children: ReactNode;
}

export default function GlobalError({ children }: ErrorProps) {
  return (
    <html>
    <body>
    {children}
    </body>
    </html>
  );
}
```

**Reply by Daniel4lm** [reply]
Doesn't work. I have under each router group (under (something) - ...) layout.tsx, page.tsx and not-found.tsx which works in case of layout but not-found.tsx simply doesn't. It only renders typical next.js 404 page

**Reply by DevQiao** [reply]
I'm suffering the same problem, where is the issue? I didn't find it, any link?
my current version of Next.js is 13.4.9, but it seems not be solved yet, sadly

**Reply by PeterdQuinn** [reply]
this annoying

**Comment by jessicarush** [comment]
In addition to `not-found.js` files not working with route groups for multiple root layouts, I have also found that they do not work for the [internationalization setup described in the docs](https://nextjs.org/docs/app/building-your-application/routing/internationalization) which uses dynamic routing.

With this setup, `not-found.js` will not work:

```
app
  ├─[lang]
  │  ├─about
  │  ├─layout.js
  │  ├─not-found.js
  │  └─page.js
  ├─favicon.ico
  └─globals.css
```

Edit: Can confirm the workaround to use a dynamic catch-all route also works with the internationalization example:

```
app
 ├─[lang]
 │  ├─[...not-found]
 │  │  └─page.js
 │  ├─about
 │  ├─layout.js
 │  ├─not-found.js
 │  └─page.js
 ├─favicon.ico
 └─globals.css
```

**Reply by rocerchen** [reply]
if use [middleware ](https://nextjs.org/docs/app/building-your-application/routing/internationalization)
and change middleware.js code to "NextResponse.rewrite" that http://localhost:3000/ will show default local content

// e.g. incoming request is /products
// return /en-US/products html
return NextResponse.rewrite(
      new URL(
        `/${locale}${pathname.startsWith("/") ? "" : "/"}${pathname}`,
        request.url
      )
);

 app
 ├─[lang]
 │  ├─[...not-found]
 │  │  └─page.js

import { notFound } from 'next/navigation';
export default function NotFoundDummy() {
  notFound();
}`

show notFound() is NEXT_NOT_FOUND error

**Comment by ginifizz** [comment]
I juste managed to get my not-found route working in a group layout using the trick mentioned here: 
https://stackoverflow.com/a/75625136
and using a "dynamic catch all route" at the root of my group layout which just calls "notFound()" when a page doesn't match any route. It's ugly, but it works for now.

**Reply by victorandree** [reply]
Interesting approach. I tried it out and it works if I do this:

```plain
app
├── (app1)
│   ├── [...not-found]
│   │   └── page.tsx
│   ├── layout.tsx
│   ├── not-found.tsx
│   └── page.tsx
└── (app2)
    └── admin
        ├── layout.tsx
        └── page.tsx
```

In `app/(app1)/[...not-found]/page.tsx`, I do this:

```typescript
import { notFound } from 'next/navigation';

export default function NotFoundDummy() {
  notFound();
}
```

This seems to work: The catch-all route raises `notFound`, which is then caught by the `app/(app1)/not-found.tsx`. I'll update the original issue with this as a workaround.

**Reply by Daniel4lm** [reply]
Of course, I tried [...not-found] in the application router (outside of grouped routes) and it works, but that seems like some sort of workaround. Also if you create layout.tsx(only barebone with jtml and body) and not-found.tsx in the app router(also outside of grouped routes) it will render 404 page

**Comment by surjithctly** [comment]
I confirm this does not work and created a [Stackblitz Reproduction](https://stackblitz.com/edit/stackblitz-starters-nan4gk?file=app%2F(website)%2Fnot-found.tsx)

**Reply by markudevelop** [reply]
same with dynamic locale param

**Comment by Daniel4lm** [comment]
Well, most of this functionality related to not-found.tsx modules is now broken in canary version 13.4.10. Only root not-found.tsx without CSS styles is rendered. I was on ver. 13.4.12 and reverted back to 13.4.9 until this issue is resolved.

**Reply by lucashfreitas** [reply]
+1 on this. had to rollback to 13.4.9 as not-found inside route groups is not working anymore.

**Reply by JunkyDeLuxe** [reply]
Confirm i reproduce the exact same problems like Daniel4lm, only root not-found.tsx without CSS styles is rendered, furthemore, the return notFound() method in a segment leads to Error: notFound() is not allowed to use in root layout

**Reply by victorandree** [reply]
I can confirm that this appears broken, and have updated the original discussion to note this.

**Reply by victorandree** [reply]
Related issue: https://github.com/vercel/next.js/issues/53694. The issue references this discussion, but it seems like discussions don't automatically get a reference back to the issue. The issue has an open PR, so hopefully this can be resolved soon (I still consider needing `[...not-found]/page.tsx` a hack, though).

**Reply by Daniel4lm** [reply]
> Related issue: https://github.com/vercel/next.js/issues/53694. The issue references this discussion, but it seems like discussions don't automatically get a reference back to the issue. The issue has an open PR, so hopefully this can be resolved soon (I still consider needing `[...not-found]/page.tsx` a hack, though).

Sure, but it seems the `[...not-found]` hack doesnt work either correctly in latest versions. Ver. 13.4.9 works fine.

**Reply by victorandree** [reply]
@Daniel4lm Since #53694 is not resolved, I would not expect this to work on `next@latest` or `next@canary`, yet. I think once it is resolved (i.e. merged and released), the workaround will work again.

**Reply by victorandree** [reply]
[v13.4.14-canary.3](https://github.com/vercel/next.js/releases/tag/v13.4.14-canary.3) fixes #53694 via #53703. I've tested my reproduction with `next@canary`, and the workaround works for me, again.

**Reply by Daniel4lm** [reply]
@victorandree  Nice man. I saw that PR also but wasn't sure. I think v13.4.14 should be out very soon so we can surely jump to new ver.

**Comment by Daniel4lm** [comment]
I can confirm that not-found rendering inside route groups works again in v.13.4.16.

**Comment by J4v4Scr1pt** [comment]
I'm on Version 13.4.16, still not working for me.. 😑
If I go to /NoRoute I get the standard 404 page not mine.
![image](https://github.com/vercel/next.js/assets/5929055/8527f141-a894-40c2-871b-6c63fd143cb2)

**Reply by victorandree** [reply]
You still need to have a "catch-all" route to capture and raise the not found (e.g. `app/(main-app)/[[...not-found]]/page.tsx`). There is otherwise no way for Next.js to know "which" not-found.tsx file to use. See the original post in this discussion.

**Reply by J4v4Scr1pt** [reply]
Ahh ok I see, Tack! 😁

**Reply by J4v4Scr1pt** [reply]
And that did the trick 🥳!

But will this always be the trick or is this a temp solution? 🤔

**Reply by Daniel4lm** [reply]
I believe it's because you have two separated route groups(with its own layouts and not-found pages respectively) and there is no other root layout besides them. If you navigate to any wrong path like /Overview/none or /TestRoute/abc  you should get your not-found page from (main-app) but something like /NoRoute  is out of the scope(don't match any of your groups). Thats's why it renders Next.js default 404 page. With catch-all you bypass this problem.

**Comment by vignesh-gupta** [comment]
@victorandree any updates? is it solved?

**Reply by victorandree** [reply]
No, I don't have any updates (FYI: I'm not associated with Next.js). I don't know if this will be "solved" or how it would be solved. The workaround listed in the original post does work, even if it feels like a hack (and is not documented as being supported).

**Comment by vignesh-gupta** [comment]
I think the best guys for this would be @icyJoseph . Could you please drop some advice?

**Reply by vignesh-gupta** [reply]
I've create a similar discussion  #54603

**Comment by LouisJackson** [comment]
Tried to apply the dynamic catch-all technique, it works great on localhost, but not on Vercel. I get a 500 error on Vercel.

`Error: Cannot find module '/var/task/nextjs/.next/server/app/(global)/page.js'
`

<img width="190" alt="Screenshot 2023-09-06 at 13 33 24" src="https://github.com/vercel/next.js/assets/8021023/6071b3cc-b9e3-4509-85c4-dea492f09e5a">

Any ideas?

**Reply by vignesh-gupta** [reply]
If you put not-found outside (global). It should work

**Comment by bwmirek** [comment]
![image](https://github.com/vercel/next.js/assets/10122893/a9001991-c133-4d42-b87f-d8fb37ded5ac)

Tried to put `not-found.tsx` in different route group to force using specific `layout.tsx`, not working in Next.js 14.0.4.

**Comment by J4v4Scr1pt** [comment]
When using the solution with a catch-all-route for notFound. Parallel routing with only a layout file breaks and tries to render the catch-all-route. Workaround is to add page.tsx that returns null...

There is a lot of workarounds with Nextjs.. hope this do not become a pattern...

**Reply by sungjun-finda** [reply]
I also has this problem too...

**Reply by eunjeechoi** [reply]
I'm suffering the same problem

**Reply by edssv** [reply]
Replacing [...not-found] with [[...not-found]] solved this problem.

**Comment by cini9** [comment]
I'm working with Next v14.0.2 and this still seems to be an issue. I've tried the catch-all technique with my internationlized app as shown below and it works as long as the URL takes into account the "lng". For example 

localhost/fr/abc routes to my not-found page ✅ 
localhost/abc does not route to my not-found page ❌ 

![Screenshot 2024-02-13 at 08 58 16](https://github.com/vercel/next.js/assets/56114997/df95e7bf-68df-41e2-ac19-3de733276f25)


If I put the catch-all route outside of [lng] folder, neither of them works. 

Is there another workaround that I'm missing to get this to work ?

**Reply by NicolasHov** [reply]
Spent the day trying to figure it out also... 😭

**Reply by jeltehomminga** [reply]
you need to have logic in your middleware that every path without a locale should redirect to locale found on header, cookie or otherwise default locale. Then localhost/abc should redirect to localhost/fr/abc and it will work

check the docs for the i18n solution you're using 
for me it looks like this in middleware.ts

`  let lng
  if (req.cookies.has(cookieName)) lng = acceptLanguage.get(req.cookies.get(cookieName)?.value)
  if (!lng) lng = acceptLanguage.get(req.headers.get('Accept-Language'))
  if (!lng) lng = fallbackLng

  const localePath = `/${lng}`
  const pathnameHasLocale = languages.some((loc) => req.nextUrl.pathname.startsWith(`/${loc}`))
  if (pathnameHasLocale) return
  if (!(req.nextUrl.pathname === localePath || pathnameHasLocale)) {
    const redirectUrl = `${req.nextUrl.origin}${req.nextUrl.basePath}${localePath}${req.nextUrl.pathname}`
    // Redirect to the expected path if it's not the current path
    return NextResponse.redirect(redirectUrl)
  }


`

**Comment by morganney** [comment]
This is still not supported as far as I can tell on v14.1.0 . If you use multiple root layouts, good luck on a custom 404 page. Wasting time on a custom 404 page is madness.

**Comment by TommySorensen** [comment]
I'm pretty sure all you need to make `notfound.js` work is `layout.js` file that returns children.
Take a look at the example from the next-intl package that use the same workaround just with a [locale] folder.

```ts
// /app/layout.tsx
import {ReactNode} from 'react';

type Props = {
  children: ReactNode;
};

// Since we have a `not-found.tsx` page on the root, a layout file
// is required, even if it's just passing children through.
export default function RootLayout({children}: Props) {
  return children;
}
```

Check repo here https://github.com/amannn/next-intl/blob/main/examples/example-app-router-playground/src/app/layout.tsx

**Reply by goodallman** [reply]
This solution is good only if you don't need anything from the layout that is supposed to display the notFound message :/ If you need the layout to be shown, bad luck.

**Reply by kbn4191** [reply]
This worked for me

**Reply by tomtitherington** [reply]
Worked for me too, thanks! 

Just a heads up for anyone using this technique, make sure you import your Metadata and CSS imports into the root `app/not-found.tsx/jsx` file (or this new `layout.tsx/js`) as they will not be included otherwise. I made this mistake and was wondering why my TailwindCSS classes weren't working...

**Reply by iwasrobbed** [reply]
This makes the most sense since layouts are nested by default. So what I did was:
- `app/layout.tsx` like above
- `app/not-found.tsx` and wrapped that in one of my `(group)/layout.tsx` layouts that had the HTML and body tags in it

e.g. in `app/not-found.tsx`:

```typescript
import PlatformLayout from './(platform)/layout'

export default function NotFound() {
  return (
    <PlatformLayout>
      /* not found page code here */
    </PlatformLayout>
  )
}

```

Then that works great for all other groups like `(group1)` and `(group2)` so the `app/not-found.tsx` is properly shared

**Comment by Diwanshumidha** [comment]
It Still Not Work for me in next@14.0.4 +1

**Comment by henokyehulu** [comment]
Same here 😭 NotWorking()

version -> 14.1.4

**Comment by shunshimono** [comment]
Next.js 14.2.3 has the same problem. Moreover, using 'import { notFound } from 'next/navigation'' does not return a 404 HTTP status code. This is truly critical.

**Comment by YYGod0120** [comment]
This issue still exists, I am using v14.0.2. The `app/[lan]/not-found.tsx` still doesn't work. However, the catch-all method can achieve the same effect, but I still hope to get this resolved. 😔

**Comment by jeltehomminga** [comment]
Same issue for me, unbelievable it is so hard to be able to show a custom 404 page if you have i18n in your app

**Comment by jpmelnik** [comment]
SImilar issue for me, I can't add a custom layout in [...not-found].

**Comment by taheramr** [comment]
any updates on this?

**Comment by Oliver-Turp** [comment]
Having this same issue myself on v14.2.4

**Comment by Diwanshumidha** [comment]
any updates ???

**Reply by nextjsdude** [reply]
look at my solution that worked in my case: [-9795255](https://github.com/vercel/next.js/discussions/50034#discussioncomment-9795255)

**Comment by nextjsdude** [comment]
i faced this issue for about one month. But finally fixed. This equally caused hydration errors in my NextJS app. here's what i did: 

1. i added the root layout.tsx to the root of my app. i know the Next docs says it should be removed, but i added it anyways.
2. Only one layout.tsx ( the parent/root layout.tsx ) can have the  html and body elements, so i Configure the different layouts, to fit that. Thats it.

**Comment by spacecat** [comment]
I will not implement this now because none of the work-arounds seem reliable IMO, instead add this to the backlog. But to my future self:

1. Read everything after this comment.
2. Check if there is a better work-around.
3. Check if there is an official fix.
4. Check if this issue has been closed and merged; if so update next.js NPM package and implement this in your app.

**Reply by morganney** [reply]
0. Consider another framework.

**Comment by CarlosVergikosk** [comment]
bump?

**Comment by CiobanuMarius9** [comment]
I am trying for months to find a good fix for this.

The main problem for me is this [...not-found]/page.js, SEO wise all the invalid routes return a 200 because the not-found catch all route is a valid route, this causes Soft 404 errors on google search console.

Removing the catch all route will result in only redirecting to not-found.js when calling notFound() which won't cover all the cases so it will default to the black next js 404 page.

Did someone find a solution to use this workaround and also change the status code to 404 ?


import { notFound } from 'next/navigation';

export const metadata = {
  robots: {
    index: false,
    follow: false,
  },
};

export default function NotFoundDummy() {
  notFound();
}

**Comment by bryce-by** [comment]
The proposed workaround does not work if your application requires a catch-all at the root. These will conflict.

`Error: You cannot use different slug names for the same dynamic path`

```
app
├── (app1)
│   ├── [...slug]
│   ├── [...not-found]
│   │   └── page.tsx
│   ├── layout.tsx
│   ├── not-found.tsx
│   └── page.tsx
└── (app2)
    ├── admin
    │   └── page.tsx
    └── layout.tsx
```

An optional catch all will also result in an error:

`Error: You cannot use both an required and optional catch-all route at the same level`
```
app
├── (app1)
│   ├── [[...slug]]
│   ├── [...not-found]
│   │   └── page.tsx
│   ├── layout.tsx
│   ├── not-found.tsx
│   └── page.tsx
└── (app2)
    ├── admin
    │   └── page.tsx
    └── layout.tsx
```

**Reply by sameerxanand** [reply]
I have this problem (referencing the second example w/ the optional catch all). @leerob Any thoughts on how to work around this?

**Comment by AudereSemper** [comment]
bump, i've the same problem too.

**Comment by fxckdead** [comment]
bump, this still happens on next 15.0.2 💡

**Comment by jjdev97** [comment]
Hi everyone,

I'm working with Next.js 15.0.3 and wanted to share a potential solution to the issue regarding not-found.tsx and group-specific layouts.

As I’m not a professional, please consider this a conceptual approach and use it with caution. Feedback is welcome!

My App Structure:

app
├── (main)
│   ├── [...not-found]
│   │   └── page.tsx
│   ├── layout.tsx
│   ├── not-found.tsx
│   └── page.tsx
└── (studio)
    │   └── page.tsx
    └── layout.tsx

**Steps to Implement:**

**1. Create a Custom Catch-All Route for Each Group**
Add a catch-all route in the (main) group to trigger notFound().
```
// src/app/(main)/[...not-found]/page.tsx
import { notFound } from "next/navigation";

export default function CatchAll() {
  notFound(); // Triggers Next.js to send a 404 status, important for SEO
}
```
**2. Add not-found.tsx in Each Group**
Define a not-found.tsx file for group-specific 404 pages.
```
// src/app/(main)/not-found.tsx
export default function MainNotFound() {
  return (
    <div>
      <h1>404 - Main Area</h1>
      <p>The page you’re looking for does not exist.</p>
    </div>
  );
}
```

**Why This Works:**

The notFound() function ensures that unmatched routes return a proper 404 HTTP status, which is essential for SEO.
Adding not-found.tsx in each group allows for group-specific error handling and layouts.
Works seamlessly in both development and production (tested on Vercel).

I hope this helps others! As mentioned earlier, please use this code with caution, and feel free to share any improvements or feedback.

**Comment by spacecat** [comment]
@leerob / @vercel: It would be nice if you could put labels on the issues that are the most commented/viewed/popular. Maybe these already exist in that case disregard this post. If not then I am thinking it would be nice to see a label on each of these issues. For example you could add "In Progress",  "Releasing in Q1 2025", "In Roadmap", or something similar... just so that we as devs know when to expect these features.

**Comment by lior-amsalem** [comment]
the only workaround i've found was to have the top level layout with basic <html>{children}</html> and the error at the same level and having sub level layouts

in some kind of hope that it won't render HTML tag twice and be weird in the final output


subscribing to follow this issue

**Comment by jademh** [comment]
+1 following this issue

**Comment by shockbs** [comment]
bump

**Comment by hunghuy201280** [comment]
+1 following this

**Comment by lovrozagar** [comment]
+1

**Comment by philwolstenholme** [comment]
This looks relevant, I think? https://x.com/huozhi/status/1921690791694446919

> When will you need it?
> - **You have multiple root layouts for different groups**
> - You can't have the root layout on the very first top segments layer

Available in v15.4.0-canary.30 and up.

**Reply by sebastian-nowak** [reply]
It doesn't fix the issue of nested `not-found.tsx` files not being picked up automatically. The `[...not-found]/page.tsx` workaround is still necessary.

**Comment by Mylothros** [comment]
If you want an SSG website with a custom 404 page, you should:

Remove export const dynamic = "force-static".
This allows Next.js to fallback to SSR for routes not statically generated.

Keep your generateStaticParams() function.
This tells Next.js which pages to pre-build and serve as static (SSG).

Leave dynamicParams at its default value (true). No need to explicitly add it.
This enables Next.js to handle dynamic routes not in your static params via SSR fallback.

Inside your page component, check if the data for the given params exists.
If it doesn’t, call notFound() to render your custom 404 page (app/not-found.tsx).

This way:

Known pages are served statically with great performance.

Unknown pages are server-rendered on demand and generating your custom 404 page

**Comment by jj-ivx** [comment]
How is this not a solved issue? Apparently these 2 features cannot be used together and there is no mention of it in the docs.

**Reply by icyJoseph** [reply]
You can use, https://nextjs.org/docs/app/api-reference/file-conventions/not-found#global-not-foundjs-experimental 

Though I can see the docs gap. IIRC, the problem is that if you have:

```tree
app
├── (app1)
│   ├── layout.tsx
│   └── page.tsx
└── (admin)
    └── admin
        ├── layout.tsx
        └── page.tsx
```

Then you have two root layouts, and adding `app/not-found.tsx`, means that `app/not-found.tsx` has no layout defining its `html`, `body` tags. You'd have to duplicate them into the `(app1)` and `(admin)` groups. Re-exporting can get you there but docs should mention it, and I do believe there might be a bug there?

The `global-not-found` file, requires you to define the `html` and `body` tags. Which means it wouldn't need an `app/layout.tsx` 

I am investigating a bit more though.

**Reply by jj-ivx** [reply]
Ah okay nice. That works well. I was searching the docs for "multiple root layouts" but I didn't find this. Thanks.

**Reply by stefanprobst** [reply]
@icyJoseph should this work: https://github.com/vercel/next.js/issues/86095

**Reply by icyJoseph** [reply]
Right that's the part that I wanted to investigate further, let me try to analyze it.
